spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AsyncAmqpOutboundGateway.java

```diff
@@ -127,30 +126,7 @@ public void onFailure(Throwable ex) {
 				}
 			}
 			else {
-				sendErrorMessage(exceptionToSend, this.requestMessage.getHeaders().getErrorChannel());
-			}
-		}
-
-		private void sendErrorMessage(Throwable ex, Object errorChannel) {
-			Throwable result = ex;
-			if (!(ex instanceof MessagingException)) {
-				result = new MessageHandlingException(this.requestMessage, ex);
-			}
-			if (errorChannel == null) {
-				logger.error("Async exception received and no 'errorChannel' header exists; cannot route "
-						+ "exception to caller", result);
-			}
-			else {
-				try {
-					sendOutput(new ErrorMessage(result), errorChannel, true);
-				}
-				catch (Exception e) {
-					Exception exceptionToLog = e;
-					if (!(e instanceof MessagingException)) {
-						exceptionToLog = new MessageHandlingException(this.requestMessage, e);
-					}
-					logger.error("Failed to send async reply", exceptionToLog);
-				}
+				sendErrorMessage(this.requestMessage, exceptionToSend);
```

Comment:
Since I don't see in the http://docs.spring.io/spring-integration/docs/4.3.0.BUILD-SNAPSHOT/reference/html/amqp.html#amqp-async-outbound-gateway any general error handling description, I think that should link to the `<<>>`, at least.
Although I see that `requires-reply` should be improved for the `errorChannel` bean anyway.
WDYT?